src/java.base/share/classes/sun/security/pkcs/SignerInfo.java

```diff
@@ -184,8 +193,7 @@ public SignerInfo(DerInputStream derin, boolean oldStyle)
         } else {
             // check if set of unauth attributes (implicit tag) is provided
             // (unauth attributes are OPTIONAL)
-            if (derin.available() != 0
-                && (byte)(derin.peekByte()) == (byte)0xA1) {
+            if (derin.seeOptionalContextSpecific(1)) {
                 unauthenticatedAttributes =
                     new PKCS9Attributes(derin, true);// ignore unsupported attrs
             }
```

Comment:
Later, there is extra bytes under // All done comment, changing that to use atEnd() call? Somehow I cannot add comment to that line? It's not close enough to the lines that you changed.

Reply:
Good idea.